diff --git a/custom_add_another.module b/custom_add_another.module
index e3103d7..486fc4f 100644
--- a/custom_add_another.module
+++ b/custom_add_another.module
@@ -131,20 +131,33 @@ function custom_add_another_field_widget_form_alter(&$element, FormStateInterfac
 function custom_add_another_preprocess_field_multiple_value_form(&$variables) {
   foreach (Element::children($variables['element']) as $child) {
     $child_element = &$variables['element'][$child];
-    if (isset($child_element['#custom_add_another_value']) || isset($child_element['#custom_remove'])) {
-      if (isset($child_element['#custom_add_another_value']) && isset($variables['element']['add_more']['#value']) && $variables['element']['add_more']['#value'] != t($child_element['#custom_add_another_value'])) {
-        $variables['element']['add_more']['#value'] = t($child_element['#custom_add_another_value']);
+
+    if (isset($child_element['#custom_add_another_value'])) {
+      if (isset($variables['element']['add_more'])) {
+        if (isset($variables['element']['add_more']['#value'])) {
+          $variables['element']['add_more']['#value'] = t($child_element['#custom_add_another_value']);
+        }
+        else {
+          $buttons = array_filter($variables['element']['add_more'], function($k) {
+            return strpos($k, "add_more") === 0;
+          }, ARRAY_FILTER_USE_KEY);
+          $variables['button'][key($buttons)]['#value'] = t($child_element['#custom_add_another_value']);
+        }
       }
-      if (isset($child_element['#custom_add_another_value']) && isset($variables['button']['#value']) && $variables['button']['#value'] != t($child_element['#custom_add_another_value'])) {
+      elseif (isset($variables['button']['#value'])) {
         $variables['button']['#value'] = t($child_element['#custom_add_another_value']);
       }
-      if (isset($child_element['#custom_remove']) && isset($child_element['remove_button']['#value'])) {
+    }
+
+    if (isset($child_element['#custom_remove'])) {
+      if (isset($child_element['remove_button']['#value'])) {
         $child_element['remove_button']['#value'] = t($child_element['#custom_remove']);
       }
     }
   }
 }
 
+
 /**
  * Process element callback for multiple Managed file widgets.
  */
