diff --git a/page_manager_ui/src/Controller/PageManagerController.php b/page_manager_ui/src/Controller/PageManagerController.php
index ebb08a7..3446520 100644
--- a/page_manager_ui/src/Controller/PageManagerController.php
+++ b/page_manager_ui/src/Controller/PageManagerController.php
@@ -16,7 +16,7 @@ use Drupal\Core\Url;
 use Drupal\ctools\Form\AjaxFormTrait;
 use Drupal\page_manager\PageInterface;
 use Drupal\page_manager\PageVariantInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Request;
 
@@ -58,7 +58,7 @@ class PageManagerController extends ControllerBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -73,8 +73,8 @@ class PageManagerController extends ControllerBase {
    *   The variant manager.
    * @param \Drupal\Core\Plugin\Context\ContextHandlerInterface $context_handler
    *   The context handler.
-+   * @param \Drupal\user\SharedTempStoreFactory $tempstore
-+   *   The tempstore factory.
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
+   *   The tempstore factory.
    */
   public function __construct(BlockManagerInterface $block_manager, PluginManagerInterface $condition_manager, PluginManagerInterface $variant_manager, ContextHandlerInterface $context_handler, SharedTempStoreFactory $tempstore) {
     $this->blockManager = $block_manager;
@@ -93,7 +93,7 @@ class PageManagerController extends ControllerBase {
       $container->get('plugin.manager.condition'),
       $container->get('plugin.manager.display_variant'),
       $container->get('context.handler'),
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
diff --git a/page_manager_ui/src/Form/PageReorderVariantsForm.php b/page_manager_ui/src/Form/PageReorderVariantsForm.php
index 047e7e1..aed7751 100644
--- a/page_manager_ui/src/Form/PageReorderVariantsForm.php
+++ b/page_manager_ui/src/Form/PageReorderVariantsForm.php
@@ -11,7 +11,7 @@ use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\page_manager\Entity\PageVariant;
 use Drupal\page_manager\PageInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -22,14 +22,14 @@ class PageReorderVariantsForm extends FormBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
   /**
    * Constructs a new DisplayVariantAddForm.
    *
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The tempstore factory.
    */
   public function __construct(SharedTempStoreFactory $tempstore) {
@@ -41,7 +41,7 @@ class PageReorderVariantsForm extends FormBase {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
diff --git a/page_manager_ui/src/Form/PageVariantAddForm.php b/page_manager_ui/src/Form/PageVariantAddForm.php
index 847c7fb..c47484a 100644
--- a/page_manager_ui/src/Form/PageVariantAddForm.php
+++ b/page_manager_ui/src/Form/PageVariantAddForm.php
@@ -12,7 +12,7 @@ use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\page_manager\Entity\PageVariant;
 use Drupal\page_manager\PageInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -30,7 +30,7 @@ class PageVariantAddForm extends FormBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -51,7 +51,7 @@ class PageVariantAddForm extends FormBase {
   public static function create(ContainerInterface $container) {
     return new static(
       $container->get('plugin.manager.display_variant'),
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
diff --git a/page_manager_ui/src/Form/PageVariantDeleteForm.php b/page_manager_ui/src/Form/PageVariantDeleteForm.php
index 1b242e2..5c4f70a 100644
--- a/page_manager_ui/src/Form/PageVariantDeleteForm.php
+++ b/page_manager_ui/src/Form/PageVariantDeleteForm.php
@@ -10,7 +10,7 @@ namespace Drupal\page_manager_ui\Form;
 use Drupal\Core\Form\ConfirmFormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -21,14 +21,14 @@ class PageVariantDeleteForm extends ConfirmFormBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
   /**
    * Constructs a PageVariantDeleteForm.
    *
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The tempstore factory.
    */
   public function __construct(SharedTempStoreFactory $tempstore) {
@@ -40,7 +40,7 @@ class PageVariantDeleteForm extends ConfirmFormBase {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
diff --git a/page_manager_ui/src/Form/ParameterEditForm.php b/page_manager_ui/src/Form/ParameterEditForm.php
index fa9cbec..defd3bd 100644
--- a/page_manager_ui/src/Form/ParameterEditForm.php
+++ b/page_manager_ui/src/Form/ParameterEditForm.php
@@ -13,7 +13,7 @@ use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\TypedData\PrimitiveInterface;
 use Drupal\Core\TypedData\TypedDataManagerInterface;
 use Drupal\page_manager\PageInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -50,7 +50,7 @@ class ParameterEditForm extends FormBase {
   protected $typedDataManager;
 
   /**
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -73,7 +73,7 @@ class ParameterEditForm extends FormBase {
    *   The entity type repository.
    * @param \Drupal\Core\TypedData\TypedDataManagerInterface $typed_data_manager
    *   The typed data manager.
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The temporary store.
    */
   public function __construct(EntityTypeRepositoryInterface $entity_type_repository, TypedDataManagerInterface $typed_data_manager, SharedTempStoreFactory $tempstore) {
@@ -89,7 +89,7 @@ class ParameterEditForm extends FormBase {
     return new static(
       $container->get('entity_type.repository'),
       $container->get('typed_data_manager'),
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
diff --git a/page_manager_ui/src/Form/VariantPluginAddBlockForm.php b/page_manager_ui/src/Form/VariantPluginAddBlockForm.php
index d6c9f88..632cfec 100644
--- a/page_manager_ui/src/Form/VariantPluginAddBlockForm.php
+++ b/page_manager_ui/src/Form/VariantPluginAddBlockForm.php
@@ -10,7 +10,7 @@ namespace Drupal\page_manager_ui\Form;
 use Drupal\Component\Plugin\PluginManagerInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\page_manager\PageVariantInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Request;
 
@@ -29,6 +29,8 @@ class VariantPluginAddBlockForm extends VariantPluginConfigureBlockFormBase {
   /**
    * Constructs a new VariantPluginFormBase.
    *
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
+   *   The temporary store.
    * @param \Drupal\Component\Plugin\PluginManagerInterface $block_manager
    *   The block manager.
    */
@@ -42,7 +44,7 @@ class VariantPluginAddBlockForm extends VariantPluginConfigureBlockFormBase {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore'),
+      $container->get('tempstore.shared'),
       $container->get('plugin.manager.block')
     );
   }
diff --git a/page_manager_ui/src/Form/VariantPluginConfigureBlockFormBase.php b/page_manager_ui/src/Form/VariantPluginConfigureBlockFormBase.php
index 59de6c6..abd353c 100644
--- a/page_manager_ui/src/Form/VariantPluginConfigureBlockFormBase.php
+++ b/page_manager_ui/src/Form/VariantPluginConfigureBlockFormBase.php
@@ -14,7 +14,7 @@ use Drupal\Core\Plugin\Context\Context;
 use Drupal\Core\Plugin\ContextAwarePluginAssignmentTrait;
 use Drupal\Core\Plugin\ContextAwarePluginInterface;
 use Drupal\page_manager\PageVariantInterface;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -27,7 +27,7 @@ abstract class VariantPluginConfigureBlockFormBase extends FormBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -48,7 +48,7 @@ abstract class VariantPluginConfigureBlockFormBase extends FormBase {
   /**
    * Constructs a new VariantPluginConfigureBlockFormBase.
    *
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The tempstore factory.
    */
   public function __construct(SharedTempStoreFactory $tempstore) {
@@ -60,7 +60,7 @@ abstract class VariantPluginConfigureBlockFormBase extends FormBase {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
@@ -76,7 +76,8 @@ abstract class VariantPluginConfigureBlockFormBase extends FormBase {
   /**
    * Get the tempstore.
    *
-   * @return \Drupal\user\SharedTempStore
+   * @return \Drupal\Core\TempStore\SharedTempStore
+   *   The temporary store.
    */
   protected function getTempstore() {
     return $this->tempstore->get($this->getTempstoreId());
diff --git a/page_manager_ui/src/Form/VariantPluginContentForm.php b/page_manager_ui/src/Form/VariantPluginContentForm.php
index 3d67314..bd10efd 100644
--- a/page_manager_ui/src/Form/VariantPluginContentForm.php
+++ b/page_manager_ui/src/Form/VariantPluginContentForm.php
@@ -11,7 +11,7 @@ use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
 use Drupal\ctools\Form\AjaxFormTrait;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -24,14 +24,14 @@ class VariantPluginContentForm extends FormBase {
   /**
    * Tempstore factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
   /**
    * Constructs a new VariantPluginContentForm.
    *
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The tempstore factory.
    */
   public function __construct(SharedTempStoreFactory $tempstore) {
@@ -43,7 +43,7 @@ class VariantPluginContentForm extends FormBase {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
@@ -59,7 +59,8 @@ class VariantPluginContentForm extends FormBase {
   /**
    * Get the tempstore.
    *
-   * @return \Drupal\user\SharedTempStore
+   * @return \Drupal\Core\TempStore\SharedTempStore
+   *   The temporary store.
    */
   protected function getTempstore() {
     return $this->tempstore->get($this->getTempstoreId());
diff --git a/page_manager_ui/src/Form/VariantPluginDeleteBlockForm.php b/page_manager_ui/src/Form/VariantPluginDeleteBlockForm.php
index b451074..51f14ed 100644
--- a/page_manager_ui/src/Form/VariantPluginDeleteBlockForm.php
+++ b/page_manager_ui/src/Form/VariantPluginDeleteBlockForm.php
@@ -40,10 +40,10 @@ class VariantPluginDeleteBlockForm extends ConfirmFormBase {
   /**
    * Get the tempstore.
    *
-   * @return \Drupal\user\SharedTempStore
+   * @return \Drupal\Core\TempStore\SharedTempStore
    */
   protected function getTempstore() {
-    return \Drupal::service('user.shared_tempstore')->get($this->getTempstoreId());
+    return \Drupal::service('tempstore.shared')->get($this->getTempstoreId());
   }
 
   /**
