diff --git a/src/Form/PanelizerWizardContextConfigure.php b/src/Form/PanelizerWizardContextConfigure.php
index a7ab050..27d71d2 100644
--- a/src/Form/PanelizerWizardContextConfigure.php
+++ b/src/Form/PanelizerWizardContextConfigure.php
@@ -5,7 +5,7 @@ namespace Drupal\panelizer\Form;
 use Drupal\Core\Plugin\Context\ContextInterface;
 use Drupal\ctools\ContextMapperInterface;
 use Drupal\ctools\Form\ContextConfigure;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -23,7 +23,7 @@ class PanelizerWizardContextConfigure extends ContextConfigure {
   /**
    * PanelizerWizardContextConfigure constructor.
    *
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore
    *   The shared tempstore factory.
    * @param \Drupal\ctools\ContextMapperInterface $context_mapper
    *   The context mapper.
@@ -38,7 +38,7 @@ class PanelizerWizardContextConfigure extends ContextConfigure {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('user.shared_tempstore'),
+      $container->get('tempstore.shared'),
       $container->get('ctools.context_mapper')
     );
   }
diff --git a/src/Form/PanelizerWizardContextForm.php b/src/Form/PanelizerWizardContextForm.php
index 44b6da7..7b68e7f 100644
--- a/src/Form/PanelizerWizardContextForm.php
+++ b/src/Form/PanelizerWizardContextForm.php
@@ -8,7 +8,7 @@ use Drupal\Core\Form\FormBuilderInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\TypedData\TypedDataManagerInterface;
 use Drupal\ctools\Form\ManageContext;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -24,7 +24,7 @@ class PanelizerWizardContextForm extends ManageContext {
   /**
    * The shared temp store factory.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstoreFactory;
 
@@ -35,7 +35,7 @@ class PanelizerWizardContextForm extends ManageContext {
     return new static(
       $container->get('typed_data_manager'),
       $container->get('form_builder'),
-      $container->get('user.shared_tempstore')
+      $container->get('tempstore.shared')
     );
   }
 
@@ -46,7 +46,7 @@ class PanelizerWizardContextForm extends ManageContext {
    *   The typed data manager.
    * @param \Drupal\Core\Form\FormBuilderInterface $form_builder
    *   The form builder.
-   * @param \Drupal\user\SharedTempStoreFactory $tempstore_factory
+   * @param \Drupal\Core\TempStore\SharedTempStoreFactory $tempstore_factory
    *   Shared user tempstore factory.
    */
   public function __construct(TypedDataManagerInterface $typed_data_manager, FormBuilderInterface $form_builder, SharedTempStoreFactory $tempstore_factory) {
diff --git a/src/Plugin/PanelsPattern/PanelizerPattern.php b/src/Plugin/PanelsPattern/PanelizerPattern.php
index e11167a..85b9508 100644
--- a/src/Plugin/PanelsPattern/PanelizerPattern.php
+++ b/src/Plugin/PanelsPattern/PanelizerPattern.php
@@ -6,7 +6,7 @@ use Drupal\Core\Plugin\Context\Context;
 use Drupal\Core\Plugin\Context\ContextDefinition;
 use Drupal\ctools\Context\AutomaticContext;
 use Drupal\panels\Plugin\PanelsPattern\DefaultPattern;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 
 /**
  * @PanelsPattern("panelizer")
